EventListener/ValidatorListener.php

```diff
@@ -14,23 +15,29 @@
 class ValidatorListener
 {
     private Validator $validator;
+    private ?Condition $condition;
 
-    public function __construct(Validator $validator)
-        {
-            $this->validator = $validator;
-        }
+    public function __construct(Validator $validator, ?Condition $condition = null)
+    {
+        $this->validator = $validator;
+        $this->condition = $condition;
+    }
 
     public function onKernelRequest(RequestEvent $event)
     {
         if ($event->isMasterRequest()) {
-            $this->validator->validateRequest($event->getRequest());
+            if (!($this->condition) || ($this->condition->enabled($event->getRequest()))) {
```

Comment:
Maybe you can change this to:

```suggestion
            if ($this->condition === null || ($this->condition->enabled($event->getRequest()))) {
```